altair/vegalite/v2/examples/scatter_marginal_hist.py

```diff
@@ -0,0 +1,53 @@
+"""
+Facetted Scatterplot with marginal histograms
+---------------------------------------------
+This example demonstrates how to generate a facetted scatterplot,
+with marginal facetted histograms, and how to share their respective
+- x,some y-limits.
+"""
+
+import altair as alt
+from vega_datasets import data
+
+iris = data.iris()
+
+# We do not want to specify the exact axis manually,
+# and ``alt.Bin`` expects a range, rather than a ``alt.Scale`` instance.
+xscale = alt.Scale(padding=1, zero=False)
+yscale = alt.Scale(padding=1, zero=False)
+xlim = xscale.domain
+ylim = yscale.domain
```

Comment:
I don't think this works the way you hope. at this point, ``scale.domain`` is ``Undefined``, which means it's the same as not providing ``xlim`` and ``ylim`` at all.

Reply:
Oh whoops, I assumed this to be a dynamic property, fixed